src/NixFromNpm/Npm/Types.hs

```diff
@@ -25,12 +25,33 @@ data PackageInfo = PackageInfo {
   piTags :: Record SemVer
   } deriving (Show, Eq)
 
+-- | Taken from https://nodejs.org/api/process.html#process_process_platform,
+-- and filtered to those that correspond to nixpkgs platforms.
+data NodePlatform
+  = DarwinPlatform
+  | FreeBSDPlatform
+  | OpenBSDPlatform
+  | LinuxPlatform
+  | SunOSPlatform
+  deriving (Show, Eq)
+
+-- | Parse a node platform from a string.
+parseNodePlatform :: Text -> Maybe NodePlatform
+parseNodePlatform = \case
+  "linux" -> pure LinuxPlatform
+  "darwin" -> pure DarwinPlatform
+  "freebsd" -> pure FreeBSDPlatform
+  "openbsd" -> pure OpenBSDPlatform
+  "sunos" -> pure SunOSPlatform
+  _ -> Nothing
```

Comment:
Non-blocking comment; since you're using applicative-style with `pure` why not use `empty`?